includes/managers/class-fs-clone-manager.php

```diff
@@ -0,0 +1,421 @@
+<?php
+    /**
+     * @package   Freemius
+     * @copyright Copyright (c) 2015, Freemius, Inc.
+     * @license   https://www.gnu.org/licenses/gpl-3.0.html GNU General Public License Version 3
+     * @author    Leo Fajardo (@leorw)
+     * @since     2.4.3
+     */
+
+    if ( ! defined( 'ABSPATH' ) ) {
+        exit;
+    }
+
+    class FS_Clone_Manager {
+        /**
+         * @var FS_Option_Manager
+         */
+        private $_storage;
+        /**
+         * @var array {
+         * @type bool   $is_temporary_duplicate
+         * @type int    $clone_identification_timestamp
+         * @type int    $request_handler_timestamp
+         * @type string $request_handler_id
+         * }
+         */
+        private $_data;
```

Comment:
@fajardoleo instead of accessing the data through `$this->_data['key']`, I suggest to enrich the class with the [magic methods](https://www.php.net/manual/en/language.oop5.overloading.php#object.get) `__get()`, `__set()`, `__isset()` (based on what's needed), and add the relevant properties as `@property`-ies in the PHP doc of the class for auto-complete. This way the code will be more structured (we can use auto-complete, refactoring, etc.)